ninja/errors.py

```diff
@@ -28,6 +30,22 @@ class ConfigError(Exception):
     pass
 
 
+TModel = TypeVar("TModel", bound="ParamModel")
+
+
+class ValidationErrorContext(Generic[TModel]):
+    """
+    The full context of a `pydantic.ValidationError`, including all information
+    needed to produce a `ninja.errors.ValidationError`.
+    """
+
+    def __init__(
+        self, pydantic_validation_error: pydantic.ValidationError, model: TModel
+    ):
+        self.pydantic_validation_error = pydantic_validation_error
+        self.model = model
```

Comment:
This seems like a good place to use a `@dataclass`, but I think this better matches the rest of the code style.